source/frontend/src/features/properties/map/acquisition/detail/AcquisitionOwnersSummaryContainer.tsx

```diff
@@ -0,0 +1,110 @@
+import { Api_AcquisitionFileOwner } from 'models/api/AcquisitionFile';
+import { Api_Address } from 'models/api/Address';
+import { useCallback, useEffect, useState } from 'react';
+
+import { useAcquisitionProvider } from '../hooks/useAcquisitionProvider';
+import { DetailAcquistionFileOwner } from './models';
+
+export interface IAcquisitionOwnersContainerProps {
+  acquisitionFileId: number;
+  View: React.FC<IAcquisitionOwnersSummaryViewProps>;
+}
+
+export interface IAcquisitionOwnersSummaryViewProps {
+  ownersList?: DetailAcquistionFileOwner[];
+  isLoading: boolean;
+}
+
+const AcquisitionOwnersSummaryContainer: React.FunctionComponent<
+  React.PropsWithChildren<IAcquisitionOwnersContainerProps>
+> = ({ acquisitionFileId, View }) => {
+  const [ownersDetails, setOwnersDetails] = useState<DetailAcquistionFileOwner[] | undefined>(
+    undefined,
+  );
+
+  const {
+    getAcquisitionOwners: {
+      execute: retrieveAcquisitionFileOwners,
+      loading: loadingAcquisitionFileOwners,
+    },
+  } = useAcquisitionProvider();
+
+  const fetchOwnersApi = useCallback(async () => {
+    if (acquisitionFileId) {
+      console.log(acquisitionFileId);
+      const acquisitionOwners = await retrieveAcquisitionFileOwners(acquisitionFileId);
+      console.log(acquisitionOwners);
+      if (acquisitionOwners !== undefined) {
+        const ownerDetailList = acquisitionOwners.map(o => {
+          return {
+            ownerName: getOwnerDisplayName(o),
+            ownerOtherName: o.lastNameOrCorp2?.trim(),
+            ownerDisplayAddress: getFormatedAddress(o.address),
+          } as DetailAcquistionFileOwner;
+        });
+        setOwnersDetails([...ownerDetailList]);
+      }
+    }
+  }, [acquisitionFileId, retrieveAcquisitionFileOwners]);
+
+  useEffect(() => {
+    fetchOwnersApi();
+  }, [fetchOwnersApi]);
+
+  return <View ownersList={ownersDetails} isLoading={loadingAcquisitionFileOwners} />;
+};
+
+export default AcquisitionOwnersSummaryContainer;
+
+const getOwnerDisplayName = (owner: Api_AcquisitionFileOwner): string => {
+  let nameDisplay = concatValues([owner.givenName, owner.lastNameOrCorp1]);
+  if (owner.incorporationNumber && owner.incorporationNumber.trim() !== '') {
```

Comment:
FYI we have a pattern for this where we create an array of the joinable elements, filter out the empty ones, and then do a .join that is more concise.

Reply:
Already in place, it's just the Incorporation number that needed a special transform.